staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -184,19 +210,32 @@ func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (execu
 	}
 }
 
-func (rmState *RequestManagementState) pickFlowSchema(rd RequestDigest) *rmtypesv1alpha1.FlowSchema {
+func (rmState *requestManagementState) pickFlowSchema(rd RequestDigest) *rmtypesv1alpha1.FlowSchema {
+	for _, flowSchema := range rmState.flowSchemas {
+		if matchesFlowSchema(rd, flowSchema) {
+			return flowSchema
+		}
+	}
 	return nil
-	// TODO: implement
 }
 
 // ComputeFlowDistinguisher extracts the flow distinguisher according to the given method
 func (rd RequestDigest) ComputeFlowDistinguisher(method *rmtypesv1alpha1.FlowDistinguisherMethod) string {
-	// TODO: implement
-	return ""
+	if method == nil {
+		return ""
+	}
+	switch method.Type {
+	case rmtypesv1alpha1.FlowDistinguisherMethodByUserType:
+		return rd.User.GetName()
+	case rmtypesv1alpha1.FlowDistinguisherMethodByNamespaceType:
+		return rd.RequestInfo.Namespace
+	default:
+		// this line shall never reach
+		panic("invalid flow-distinguisher method")
+	}
 }
 
 // HashFlowID hashes the inputs into 64-bits
 func hashFlowID(fsName, fDistinguisher string) uint64 {
-	// TODO: implement
-	return 0
+	return crc64.Checksum([]byte(fsName+fDistinguisher), crc64.MakeTable(crc64.ECMA))
```

Comment:
OK for an initial implementation, but note that it is not necessary to concatenate the slices; instead we can make a Table and then call `crc64.Update` a few times (I suggest three: one for fsName, one for a separator, and one for fDistinguisher).

Reply:
i will need to investigate into the differences between the two..